gamemode/modules/main_menu/cl_mainmenu_create.lua

```diff
@@ -0,0 +1,328 @@
+RVR.MainMenu = RVR.MainMenu or {}
+local mainMenu = RVR.MainMenu
+
+local textEntryBg = Material( "rvr/backgrounds/mainmenu_textbackground.png" )
+local textEntryBgShort = Material( "rvr/backgrounds/mainmenu_textbackground_short.png" )
+
+mainMenu.backgrounds = mainMenu.backgrounds or {}
+mainMenu.backgrounds[mainMenu.MENU_CREATE] = Material( "rvr/backgrounds/mainmenu_create.png" )
+
+function mainMenu.createEntry( data )
+    data.action = function( val )
+        data.tab[data.name] = val
+    end
+
+    local text = data.text
+    local entryW = data.entryWidth
+    local tooltip = data.tooltip
+    local entryClass = data.entryType
+    local entryH = data.entryHeight or ScrH() * 0.04
+
+    local x = data.x or 100
+    local y = data.y
+    if not y then
+        mainMenu.entryY = mainMenu.entryY + 0.13 * mainMenu.h
+        y = mainMenu.entryY
+    end
+
+    local container = mainMenu.container
+
+    local headerSpacing = 5
+    if data.headerSpacing then
+        headerSpacing = data.headerSpacing
+        y = y + headerSpacing - 5
+    end
+
+    local label = vgui.Create( "DLabel", container )
+    label:SetFont( "RVR_StartMenuLabel" )
+    label:SetText( text )
+    label:SetTextColor( mainMenu.darkerYellow )
+    label:SizeToContentsX()
+    label:SetTall( ScrH() * 0.04 )
+    label:SetPos( x, y - ScrH() * 0.04 - headerSpacing )
+    label:SetMouseInputEnabled( true )
+    if tooltip then label:SetTooltip( tooltip ) end
+
+    data.label = label
+
+    local errorLabel = mainMenu.createErrorLabel( data.name, 0, 0 )
+
+    function errorLabel:PerformLayout()
+        local _w = label:GetWide()
+        local _x, _y = label:GetPos()
+        self:SetPos( _x + _w + 20, _y )
+    end
+
+    local entry = vgui.Create( entryClass, container )
+    entry:SetPos( x + 10, y )
+    entry:SetSize( entryW, entryH )
+    if tooltip then entry:SetTooltip( tooltip ) end
+
+    return entry
+end
```

Comment:
Some magic numbers in here, some repeated, would be nice to get those into variables

(actually, applies to the whole file)

Reply:
Oh god no

Reply:
Made some changes, what do you think?